config/m4/cuda.m4

```diff
@@ -96,9 +101,12 @@ AS_IF([test "x$cuda_checked" != "xyes"],
                [NVCC_CFLAGS="$NVCC_CFLAGS -O0 -g"],
                [NVCC_CFLAGS="$NVCC_CFLAGS -O3 -g -DNDEBUG"])
          AS_IF([test "x$cuda_happy" = "xyes"],
-               [AS_IF([test $CUDA_MAJOR_VERSION -eq 11],
-                      [NVCC_ARCH="${ARCH8} ${ARCH9} ${ARCH10} ${ARCH11}"])
-                AC_SUBST([NVCC_ARCH], ["$NVCC_ARCH"])])
+               [AS_IF([test "x$with_nvcc_gencode" = "xdefault"],
```

Comment:
i would simplify user experience a bit:
we can add a helper fn on top of this file:
```
  AC_DEFUN([CUDA_ARCH_LIST], [
      arch_config_str=$1
      NVCC_ARCH=""
      for arch in $(echo ${arch_config_str} | tr "," " "); do
           NVCC_ARCH="${NVCC_ARCH} -gencode=arch=compute_${arch},code=sm_${arch}"
      done
   ])
```
then if with_nvcc_gencode != default:
CUDA_ARCH_LIST([$with_nvcc_gencode]) 
which will set NVCC_ARCH.

in that case user can set --with-nvcc-gencode=50,60,70
wdyt?

Reply:
i thought about it, the drawback here is that by setting ucc configure arg you explicitly compile for a particular GPU i.e. generating particular PTX and SAAS. What if user wants set only virtual arch?

Reply:
yes, i think so, if you want to build ucc binary today but support future GPU architectures you will need to add ptx